src/layout/DashboardPage.tsx

```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+
+import { useDataSource, DataSourceContext } from '@/hooks';
+import Loading from '@/components/Loading';
+import Topbar from '@/layout/Topbar';
+
+export default function DashboardPage({
+  children,
+}: React.PropsWithChildren): JSX.Element {
```

Comment:
Created this component so we have the context for QueryClient outside in the layout and here we make sure that `useDataSource` can call `useQueryClient`.